numpy/f2py/tests/test_quoted_character.py

```diff
@@ -28,3 +30,27 @@ class TestQuotedCharacter(util.F2PyTest):
                         reason='Fails with MinGW64 Gfortran (Issue #9673)')
     def test_quoted_character(self):
         assert_equal(self.module.foo(), (b"'", b'"', b';', b'!', b'(', b')'))
+
+def test_f2py_init_compile():
+    # flush through the f2py __init__
+    # compile() function code path
+    # as a crude test for input handling
+    # following migration from exec_command()
+    # to subprocess.check_output() in gh-11937
+
+    fsource =  b'''
+             integer function foo()
+             foo = 10 + 5
+             return
+             end
+	     '''
+    ret_val = numpy.f2py.compile(fsource,
+                                 modulename='return_check')
+    # check for compile success return value
+    assert_equal(ret_val, 0)
+    # check for sensible result of Fortran function
+    # that means we can import the module name in Python
```

Comment:
I feel like we already have some infrastructure set up to test this type of thing with `f2py`, but I'm unsure where

Reply:
Thanks for the tip--I found it in `numpy/f2py/tests/util.py`. Unfortunately, `build_module()` in there also implements its own logic to feed options to `f2py` through `Popen` and completely circumvents the code block I'm trying to patch (which is why it is not covered by tests in the first place).

But you're right -- I should be to steal the import trick, which uses `tempfile` and `importlib` + various helper functions in `util.py`